src/ht.c

```diff
@@ -155,14 +161,13 @@ static void internal_ht_grow(SdbHt* ht) {
 	*ht2 = swap;
 	ht_free (ht2);
 }
-#endif
 
 static bool internal_ht_insert_kv(SdbHt *ht, HtKv *kv, bool update) {
 	bool found = false;
 	if (!ht || !kv) {
 		return false;
 	}
-	ut32 bucket, hash = ht->hashfn (kv->key);
+	ut32 bucket, hash = HASHFN (ht, kv->key);
```

Comment:
hashfn should be always non-null imho, because your macro us using the k as int which is wrong hashing method (address of var != contents of var)

Reply:
`k` is used directly only when a hashfn is not provided (e.g. to simulate a dict, since you said many times you think sometimes you just want numbers as keys). In general, I do believe that most of the times you actually want a hashfn, but I see no reasons to prevent numbers as keys.

There's a small catch here about `sizeof (void *) != sizeof (ut64)` on all archs, so in a future PR I will use `ut64` as types for key/value instead of `void*` as done in `dict.` What do you think?